### ISSUE TITLE
Decorator Component's hover state is missing

### ISSUE THREAD

**Issue by DhanashreeKolhe** [carbon-design-system/ibm-security#603]
## Bug - Decorator Component's hover state is missing
 

**Expected behavior -**
Decorator's color should change on hover

**Actual behavior -**

There is no change in color of the Decorator on hover

https://ibm-security.carbondesignsystem.com/?path=/story/components-decorator--default

### Screenshots
<img width="217" alt="image" src="https://user-images.githubusercontent.com/66696144/84245164-44384e80-ab22-11ea-858e-c96030ab84b1.png">

**Version -** 
1.25.0

**Comment by SimonFinney** [comment]
Hi @DhanashreeKolhe ! The decorator will only supply a hover state when it is interactive ( https://ibm-security.carbondesignsystem.com/?path=/story/components-decorator--with-href ) - Hope this helps!

**Comment by DhanashreeKolhe** [comment]
Hi @SimonFinney We are using this component inside Data Table and need a color recommendation when the table row has hovered upon. I have taken references from several dark theme designs where Decorator is used inside a Data Table.

<img width="873" alt="image" src="https://user-images.githubusercontent.com/66696144/84256205-3938ea80-ab31-11ea-9db8-ac3f400f59e4.png">

The current color of default Decorator merges with Data table hover's background-color

<img width="876" alt="image" src="https://user-images.githubusercontent.com/66696144/84256336-64233e80-ab31-11ea-95ef-37fc7330498a.png">

**Comment by cameroncalder** [comment]
Hey @SimonFinney, yeah I asked for the team to log this issue so we could investigate a solution/track it.

Quick question @DhanashreeKolhe, is the row itself clickable? As in does it require a hover state for the row?

**Comment by cameroncalder** [comment]
^ We'll need to address the issue either way, but just curious for this particular example

**Comment by DhanashreeKolhe** [comment]
@cameroncalder This row has 'Notify' action on hover hence hover state is required
<img width="719" alt="image" src="https://user-images.githubusercontent.com/66696144/84682981-1d2fb180-af54-11ea-8209-1e1141fc8d23.png">

**Comment by SimonFinney** [comment]
@cameroncalder Is this something that's still being explored on the design side?

**Comment by cameroncalder** [comment]
> @cameroncalder Is this something that's still being explored on the design side?

We can take a look at it this week
